src/ngResource/resource.js

```diff
@@ -588,14 +657,23 @@ angular.module('ngResource', ['ng']).
               }
             });
 
+            if (!isInstanceCall && action.cancellable) {
+              var deferred = $q.defer();
+              httpConfig.timeout = deferred.promise;
+              value.$cancelRequest = deferred.resolve;
```

Comment:
I haven't given it too much thought tbh, but here is what I think:

- `httpConfig` and `value` do not reference each other, so the question is whether each one is disposed of properly on its own.
- `httpConfig` should get gc'ed (but maybe it doesn't because it is referenced inside the [promise callback][1] - I am not sure). If it doesn't, we should fix it. In any case, this is not introduced by or in any way related to changes in this PR.
- `value.$cancelRequest` does live on, so we need to de-reference `deferred` s soon as that's possible, in order to enable it to be gc'ed. That is what's done in the `finally` block.

I might have missed something though, so take the above with a grain of salt.

[1]: https://github.com/gkalpak/angular.js/blob/feat-%24resource-abortable-requests/src/ngResource/resource.js#L685